src/DataAPI.jl

```diff
@@ -93,4 +100,11 @@ struct All{T<:Tuple}
     All(args...) = new{typeof(args)}(args)
 end
 
+struct BroadcastedAll{T<:Tuple}
```

Comment:
Why it is better than just?
```
struct BroadcastedAll{T <: All}
    all::T
end
```
or
```
struct BroadcastedAll{T}
    cols::T
    BroadcastedAll(all::All{T}) where T = new{T}(all.cols)
end
```

Reply:
Not sure. I figured the first definition (nesting) would increase complexity for no advantage. The second is probably OK, though it requires creating an `All` object to create a `BroadcastedAll`. Anyway we could have two constructors.

Reply:
I am OK with any you choose - simply the one proposed here felt more natural. We can leave what you proposed I think.